tests/test_controller.py

```diff
@@ -175,32 +186,28 @@ def setup(self, _db=None):
             app.manager = self.manager
             self.controller = CirculationManagerController(self.manager)
 
-    
-class CirculationControllerTest(ControllerTest):
+    def make_default_library(self, _db):
+        return self._default_library
 
-    def setup(self):
-        super(CirculationControllerTest, self).setup()
+    def make_default_collection(self, _db, library):
+        return self._default_collection
 
-        # TODO: This is a prime candidate for optimization. A lot of
-        # tests don't need these books, and they take over 1 second to
-        # create.
```

Comment:
It's not related, but this branch implements this TODO. On my computer, creating a Work takes about 1/3 of a second. Relatively few controller tests need more than one Work, and there are about 100 controller tests in this file and in admin/test_controller.py. So this change should reduce by an average of 1 minute the time it takes the test suite to run. I haven't verified this--runtimes on Travis are highly variable--but it certainly won't hurt.